src/cmd/flux-jobspec.py

```diff
@@ -176,6 +181,197 @@ def slurm_jobspec(args):
+    return value
+
+
+def parse_fsd(s):
+    m = re.match(r".*([smhd])$", s)
+    n = float(s[:-1] if m else s)
+    unit = m.group(1) if m else "s"
+
+    if unit == "m":
+        seconds = timedelta(minutes=n).total_seconds()
+    elif unit == "h":
+        seconds = timedelta(hours=n).total_seconds()
+    elif unit == "d":
+        seconds = timedelta(days=n).total_seconds()
+    else:
+        seconds = n
+    return seconds
+
+
+def resource_walker(res):
+    while res is not None and len(res) > 0:
+        if type(res) is dict:
+            yield res
+            res = res.get("with", None)
+        elif type(res) is list:
+            res = res[0]
+
+
+def resource_type_walker(res):
+    for r in resource_walker(res):
+        yield r["type"]
+
+
+def parse_shape(shape, nslots):
+    # type: (str) -> Dict[str, Any]
+    partlist = shape.split("/")
+    if len(partlist) <= 1:
+        partlist = shape.split(">")
+
+    count_re = re.compile(
+        r"(?P<res>[^\[]+)(?:\[(?P<min>\d+)(?::(?P<max>\d+)(?::(?P<stride>\d+))?)?])?"
+    )
+
+    slot_added = False
+    # TODO: manual parsing making me grumpy, may pull in pycparser, or use ply
+    # if I can stomach it (cffi already depends on it), can't handle unit
+    # parsing for counts without doing some more work here
+    res = []
+    for r in reversed(partlist):
+        m = count_re.match(r)
+        if len(m.group("res")) < 1:
+            raise ValueError("invalid shape, no resource name in {}".format(r))
+        if m.group("min") is None:
+            count = 1
+        else:
+            count = int(m.group("min"))
+        if m.group("max") is not None:
+            count = {
+                "min": count,
+                "max": int(m.group("max")),
+                "operator": "+",
+                "operand": 1,
+            }
+            if m.group("stride") is not None:
+                count["operand"] = int(m.group("stride"))
+        if m.group("res") == "slot":
+            if m.group("min") is not None:
+                raise ValueError("slot cannot take count from shape at present")
+            slot_added = True
+            res = [create_slot("task", nslots, res)]
+        else:
+            res = [create_resource(m.group("res"), count, res)]
+    if slot_added:
+        return res
+    else:
+        return [create_slot("task", nslots, res)]
+
+
+def flux_jobspec(args):
+    # set up defaullts for options where presence matters and check invariants
+    if args.file is not None:
+        if any(
+            [
+                v is not None
+                for v in (
+                    args.nslots,
+                    args.total_tasks,
+                    args.tasks_per_slot,
+                    args.slot_shape,
+                    args.shape_file,
+                )
+            ]
+        ):
+            logger.error("--file cannot be combined with other options")
+            sys.exit(1)
+        pass  # TODO< handle file
```

Comment:
Lol.  I thought to try this functionality, but when it spit out a jobspec and didn't complain, I didn't even bother to double check the output was from the file I gave it :see_no_evil:

This should either get dropped or implemented before we merge (just making a comment so I don't click the button too quickly).